sickbeard/server/api/v2/shows.py

```diff
@@ -0,0 +1,175 @@
+
+            show_dict['genre'] = genre_list
+            show_dict['quality'] = get_quality_string(show_obj.quality)
+
+            any_qualities, best_qualities = _map_quality(show_obj.quality)
+            show_dict['quality_details'] = {'initial': any_qualities, 'archive': best_qualities}
+
+            try:
+                show_dict['location'] = show_obj.location
+            except ShowDirectoryNotFoundException:
+                show_dict['location'] = ''
+
+            show_dict['language'] = show_obj.lang
+            show_dict['name'] = show_obj.name
+            show_dict['paused'] = bool(show_obj.paused)
+            show_dict['subtitles'] = bool(show_obj.subtitles)
+            show_dict['air_by_date'] = bool(show_obj.air_by_date)
+            show_dict['flatten_folders'] = bool(show_obj.flatten_folders)
+            show_dict['sports'] = bool(show_obj.sports)
+            show_dict['anime'] = bool(show_obj.anime)
+            show_dict['airs'] = str(show_obj.airs).replace('am', ' AM').replace('pm', ' PM').replace('  ', ' ')
+            show_dict['dvdorder'] = bool(show_obj.dvdorder)
+
+            if show_obj.rls_require_words:
+                show_dict['rls_require_words'] = show_obj.rls_require_words.split(', ')
+            else:
+                show_dict['rls_require_words'] = []
+
+            if show_obj.rls_ignore_words:
+                show_dict['rls_ignore_words'] = show_obj.rls_ignore_words.split(', ')
+            else:
+                show_dict['rls_ignore_words'] = []
+
+            show_dict['scene'] = bool(show_obj.scene)
+
+            show_dict['ids']['thetvdb'] = helpers.mapIndexersToShow(show_obj)[1]
+            show_dict['ids']['imdb'] = show_obj.imdbid
+
+            show_dict['network'] = show_obj.network
+            if not show_dict['network']:
+                show_dict['network'] = ''
+
+            show_dict['status'] = show_obj.status
+
+            if try_int(show_obj.nextaired, 1) > 693595:
+                dt_episode_airs = sbdatetime.sbdatetime.convert_to_setting(
+                    network_timezones.parse_date_time(show_obj.nextaired, show_dict['airs'], show_dict['network']))
+                show_dict['airs'] = sbdatetime.sbdatetime.sbftime(dt_episode_airs, t_preset=timeFormat).lstrip('0').replace(' 0', ' ')
+                show_dict['next_ep_airdate'] = sbdatetime.sbdatetime.sbfdate(dt_episode_airs, d_preset=dateFormat)
+            else:
+                show_dict['next_ep_airdate'] = ''
+            self.api_finish(**{'show': show_dict})
+        else:
+            shows = {}
+            for show in sickbeard.showList:
+                # If self.get_argument('paused') is None: show all, 0: show un-paused, 1: show paused
+                if self.get_argument('paused', default=None) is not None and self.get_argument('paused', default=None) != show.paused:
+                    continue
+
+                indexer_show = helpers.mapIndexersToShow(show)
+
+                show_dict = {
```

Comment:
Can the creation logic for single show and multiple shows be merged? Single show and multiple shows are returning the same fields, right?